worker/src/main/java/gov/cms/ab2d/worker/processor/StreamHelperImpl.java

```diff
@@ -0,0 +1,216 @@
+package gov.cms.ab2d.worker.processor;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.nio.file.StandardOpenOption.APPEND;
+
+/**
+ * Contains the common methods for other StreamHelper implementations
+ */
+@Slf4j
+public abstract class StreamHelperImpl implements StreamHelper, AutoCloseable {
+    public enum FileOutputType {
+        NDJSON(".ndjson"),
+        NDJSON_ERROR("_error.ndjson"),
+        ZIP(".zip");
+
+        private String suffix;
+        FileOutputType(String suffix) {
+            this.suffix = suffix;
+        }
+        public String getSuffix() {
+            return suffix;
+        }
+    }
+
+    // Current file counter
+    @Getter
+    private int counter = 1;
+
+    // Passed contract number
+    @Getter
+    private final String contractNumber;
+
+    // Directory where to put the files
+    @Getter
+    private final Path path;
+
+    // Total number of bytes written to the file
+    @Getter @Setter
+    private long totalBytesWritten = 0;
+
+    // Total bytes allowed in the file
+    @Getter
+    private final long totalBytesAllowed;
+
+    // The current output stream
+    @Getter @Setter
+    private OutputStream currentStream;
```

Comment:
Would this also need to be `volatile` because it will be access in multiple threads..
And one of the threads could have closed the previous file and assigned a newly created file to this variable?